### PR TITLE
Basic application shell & layouting

### PR DESCRIPTION
**Pull request by roobscoob**
The goal of this PR is to introduce a basic application shell to later be built on. This shell includes a sidebar which supports 3 states: (Closed, Minimized, Open). The sidebar is resizable and has a button to switch between states.

An example of what the client looks like with this PR is available here: http://s3.hall.ly/rod/2024/11/scope_tMxZvqUF0o.mp4

### CODE REVIEW

File: src-tauri/src/lib.rs [hunk 1]
```diff
@@ -7,8 +9,15 @@ fn greet(name: &str) -> String {
...
+            }
+        })
         .plugin(tauri_plugin_shell::init())
         .invoke_handler(tauri::generate_handler![greet])
-        .run(tauri::generate_context!())
-        .expect("error while running tauri application");
+        .build(tauri::generate_context!())
+        .expect("error while running tauri application")
+        .run(|handle, evt| {})
 }
```
Comment (vmfunc) [inline src-tauri/src/lib.rs:23]: handle and evt are unused

```suggestion
.run(|_handle, _evt| {})
```

File: src-tauri/src/lib.rs [hunk 2]
```diff
@@ -7,8 +9,15 @@ fn greet(name: &str) -> String {
...
+                std::thread::sleep(std::time::Duration::from_nanos(1));
+            }
+        })
         .plugin(tauri_plugin_shell::init())
         .invoke_handler(tauri::generate_handler![greet])
-        .run(tauri::generate_context!())
-        .expect("error while running tauri application");
+        .build(tauri::generate_context!())
+        .expect("error while running tauri application")
+        .run(|handle, evt| {})
```
Comment (vmfunc) [inline src-tauri/src/lib.rs]: ~~resolved~~

```
        .run(|_handle, _evt| {})
```

**Review by vmfunc — APPROVED** [review]

**Review by bvvst — APPROVED** [review]
lgtm

**Comment by vmfunc** [comment]
image

**Pull request merged**